src/zookeeper/leader-election.ts

```diff
@@ -112,13 +124,9 @@ export const registerForLeaderElection = async ({
           throwDisconnectAfterThreshold(client, timeoutThreshold, cleanup);
         } else {
           // Watch the next znode down the sequence
-          watchZNodeChanges(smallestChild, currentSequence);
+          await watchZNodeChanges(smallestChild, currentSequence);
```

Comment:
await is unnecessary

Reply:
Same

Reply:
verify if it can be ignore, if so, ignore it